Sakip data not transforming cleanly

Error:

```
2019-11-04T19:06:31+00:00 ERROR Unexpected error on record <record #1 (data/sakip-sabanci/abidindino/data/abidindino-1107.xml #1), output_id:oai:cdm21044.contentdm.oclc.org:abidindino/2069>
    while executing (to_field "cho_date_range_hijri" at traject_configs/sakip_single_date_config.rb:4)
    Record: <?xml version="1.0"?>
<record xmlns="http://www.openarchives.org/OAI/2.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <header>
    <identifier>oai:cdm21044.contentdm.oclc.org:abidindino/2069</identifier>
    <datestamp>2013-09-16</datestamp>
    <setSpec>abidindino</setSpec>
  </header>
  <metadata>
    <oai_dc:dc xmlns:oai_dc="http://www.openarchives.org/OAI/2.0/oai_dc/" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/oai_dc/ http://www.openarchives.org/OAI/2.0/oai_dc.xsd">
      <dc:type>Gazete k&#xFC;p&#xFC;r&#xFC;</dc:type>
      <dc:title>Abidin Dino&#x2019;nun sergisi Paris&#x2019;te b&#xFC;y&#xFC;k ilgi g&#xF6;rd&#xFC;</dc:title>
      <dc:language>T&#xFC;rk&#xE7;e</dc:language>
      <dc:title>H&#xFC;rriyet Gazetesi.</dc:title>
      <dc:coverage>[s.l.]</dc:coverage>
      <dc:publisher>H&#xFC;rriyet Gazetesi</dc:publisher>
      <dc:date>[n.d.]</dc:date>
      <dc:description>Abidin Dino - Resim - Sanat. Haber.</dc:description>
      <dc:format>pdf</dc:format>
      <dc:identifier>http://digital.sabanciuniv.edu/abidindino/kupur/3040200003047.pdf</dc:identifier>
      <dc:rights>&#xA9; Sabanc&#x131; &#xDC;niversitesi Sak&#x131;p Sabanc&#x131; M&#xFC;zesi. T&#xFC;m haklar&#x131; sakl&#x131;d&#x131;r.</dc:rights>
      <dc:identifier>http://cdm21044.contentdm.oclc.org/cdm/ref/collection/abidindino/id/2069</dc:identifier>
    </oai_dc:dc>
  </metadata>
</record>
    Exception: NoMethodError: undefined method `+' for nil:NilClass
    /opt/traject/lib/macros/date_parsing.rb:257:in `block in hijri_range'
[ERROR] undefined method `+' for nil:NilClass
/opt/traject/lib/macros/date_parsing.rb:257:in `block in hijri_range': undefined method `+' for nil:NilClass (NoMethodError)
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer/step.rb:140:in `block in execute'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer/step.rb:135:in `each'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer/step.rb:135:in `execute'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:464:in `block (2 levels) in map_to_context!'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:504:in `handle_mapping_errors'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:463:in `block in map_to_context!'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:457:in `each'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:457:in `map_to_context!'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:582:in `block (3 levels) in process'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/thread_pool.rb:123:in `block in maybe_in_thread_pool'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'
```

--- a/lib/macros/date_parsing.rb
+++ b/lib/macros/date_parsing.rb
@@ -39,9 +39,16 @@ module Macros
             hijri_range << (to_hijri(gregorian_range.first)..to_hijri(gregorian_range.last) + 1).to_a
           end
         end
-        hijri_range.flatten!.uniq! if hijri_range.any?
-        accumulator.replace(hijri_range)
+        accumulator.replace(normalize_year_range(hijri_range))
       end
+    end
+
+    # Given an array of year values, return a flat, sorted array of unique values with `nil`s filtered out
+    # NOTE: this is not a macro, but a helper method for other macros
+    def normalize_year_range(range)
+      return range if range.empty?
+
+      range.flatten.compact.uniq.sort
     end
 
     # Takes an existing array of year integers and returns an array converted to hijri
@@ -91,8 +98,7 @@ module Macros
         accumulator.each do |val|
           range_years << ParseDate.parse_range(val) if val&.strip.present?
         end
-        range_years.flatten!.uniq! if range_years.any?
-        accumulator.replace(range_years)
+        accumulator.replace(normalize_year_range(range_years))
       end
     end
 
@@ -107,13 +113,9 @@ module Macros
         range_years = []
         accumulator.each do |val|
           range_years << val.scan(AUC_REGEX).map { |year| year.sub(AUC_DELIM, '').to_i }
-          next unless range_years.flatten!&.uniq!.nil?
-
           range_years << ParseDate.range_array(ParseDate.earliest_year(val), ParseDate.latest_year(val))
         end
-
-        range_years.flatten!&.uniq!&.sort! if range_years.any?
-        accumulator.replace(range_years)
+        accumulator.replace(normalize_year_range(range_years))
       end
     end
 

